docs/proposals/29-autoscaling-storage-volumes.md

```diff
@@ -0,0 +1,335 @@
+3. The feature flag will be promoted and eventually removed.
+
+The following metrics will be recorded before and/or after the transition of the seed to the new functionality,
+in order to evaluate the economical effect, and the reliability of the scaling solution:
+   - Average utilisation of observability volumes:
+     - Preexisting volumes (before and after)
+     - Newly created volumes (after)
+   - Percentage of observability volumes above 90% utilisation (before and after)
+   - Utilisation of observability volumes at p10, p50, p90:
+     - Preexisting volumes (before and after)
+     - Newly created volumes (after)
+   - Peak volume utilisation (after)
+   - Observability volume size: at p10, p50, p90, and average size (after)
+   - Observability volume average size and count (before)
+
+#### Enabling Autoscaling for Existing Volumes
+Autoscaling is not automatically enabled for storage volumes which already exist at the time when the new feature is
+enabled. That applies to both volumes in shoot namespaces, and those in the `garden` namespace.
+
+These volumes have large, fixed capacity, which in most cases is severely underutilised. Merely enabling autoscaling
+on them would do nothing to release the unused storage space. It is the operator's responsibility to
+decide which of those volumes to migrate to autoscaling, whether to release the unused space, and if so, whether
+to preserve the existing data on the volumes, and how.
+
+### Future Enhancements
+#### Metrics authorization
+`pvc-autoscaler` supports access control over its `/metrics` endpoint. This feature will not be utilised by the first
+implementation round. It is envisioned as a future enhancement. In that operation mode,
+the primary `pvc-autoscaler` container publishes unauthorized, plain text metrics only on the loopback interface.
+Peer pods are configured to look for `/metrics` at a different port, which is served by a secondary container, running
+the `kube-rbac-proxy`[[2]] application. It serves HTTPS and authenticates and authorizes each request against the runtime
+cluster via the TokenReview[[3]] and SubjectAccessReview[[4]] K8s APIs, before forwarding it to the internal, insecure
+loopback port.
+
+![03-runtime_structure_secure_metrics.png](assets/gep29-03-runtime_structure_secure_metrics.png)
+
+_Fig.3: Runtime structure with secure metrics_
+
+## Discussion and Limitations
+### Alternatives
+The following existing 3rd party PVC autoscalers were evaluated:
+- [topolvm/pvc-autoresizer](https://github.com/topolvm/pvc-autoresizer)
+- [lorenzophys/pvc-autoscaler](https://github.com/lorenzophys/pvc-autoscaler)
+- [DevOps-Nirvana/Kubernetes-Volume-Autoscaler](https://github.com/DevOps-Nirvana/Kubernetes-Volume-Autoscaler)
+
+**Note:** The evaluation of each autoscaling option also accounts for the utility it adds to the envisioned future
+scenario of offering PVC autoscaling service to shoot (customer) volumes.
```

Comment:
I'd like to avoid duplicating `pvc-autoscaler`'s documentation in the GEP. If the `pvc-autoscaler` documentation is incomplete, I'd rather make the necessary changes there.

On the other hand, the GEP should indeed describe the specifics of how `pvc-autoscaler` will be applied in our concrete case. The thing is, because that's basic stuff, and very obvious to me, I'm quite blind to omissions on that part. So, please take a look a the new introductory paragraph I added to the GEP (available with next push), and let me know what I'm missing.

Reply:
I agree that duplicating documentation is not great, but then again you spend 70 lines on why the alternatives are not good and we can only read between the lines why pvc-autoscaler is better.

You might have worked closer with pvc-autoscaler and hence you may already know this, but everybody else just sees you bashing a couple of projects without explicitly writing what you want/what pvc-autoscaler provides.

From my point of view, it would be a good idea to have a short section for pvc-autoscaler here as well listing what we think is relevant.

Reply:
I added a brief description of `pvc-autoscaler` in the start of the "Discussion and Limitations" section. Will be available with next push.